velero-plugins/common/shared.go

```diff
@@ -8,13 +8,30 @@ import (
 	"strings"
 
 	"github.com/fusor/ocp-velero-plugin/velero-plugins/clients"
+	"github.com/openshift/library-go/pkg/image/reference"
+	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/version"
 )
 
-func GetRegistryInfo(major, minor string) (string, error) {
+func GetRegistryInfo(major, minor string, log logrus.FieldLogger) (string, error) {
+	imageClient, err := clients.ImageClient()
+	if err != nil {
+		return "", err
+	}
+	imageStreams, err := imageClient.ImageStreams("openshift").List(metav1.ListOptions{})
+	if err == nil && len(imageStreams.Items) > 0 {
+		if value := imageStreams.Items[0].Status.DockerImageRepository; len(value) > 0 {
+			ref, err := reference.Parse(value)
+			if err == nil {
+				log.Info("[GetRegistryInfo] value from imagestream")
```

Comment:
Is this log statement supposed to include context? I may not understand how your logging works.

Reply:
I could add more to the logs, I guess -- I mainly wanted to log the method in which we're determining the name. The name itself is stored as an annotation on the resource, so it wasn't crucial for me to make sure the new code was being run. I have a feeling a lot of our logging could use some refining at some point -- some of the existing log statements are probably better as Debug vs. Info logging. For this PR, I could add the value I'm returning, but I don't know that it adds much.